tests/utils/shopper.js

```diff
@@ -46,4 +46,22 @@ export const shopper = {
 			}
 		);
 	},
+
+	selectPayment: async ( payment ) => {
+		await expect( page ).toClick(
+			'.wc-block-components-payment-method-label',
+			{
+				text: payment,
+			}
+		);
+	},
+
+	placeOrder: async () => {
+		await expect( page ).toClick( '.wc-block-components-button__text', {
+			text: 'Place Order',
+		} );
+		page.waitForNavigation( {
```

Comment:
Hi @nielslange!

The `page.waitForNavigation()` function returns a Promise, so it should be preceded by the `await` keyword to get resolved. And since it resolves when the page navigates to a new URL or reloads, we should use the `Promise.all()` method to work properly, like so:

```js
placeOrder: async () => {
  await Promise.all( [
	  page.waitForNavigation( {
		  waitUntil: 'networkidle0',
	  } ),
	  expect( page ).toClick( '.wc-block-components-button__text', {
		  text: 'Place Order',
	  } ),
  ] );
},
```

This way `page.waitForNavigation()` & `expect( page ).toClick()` will execute at the same time. The latter will resolve when the element is clicked, and the first one will resolve when the page navigates. So, executing at the same time is important in this particular case.

Otherwise, if we have this code instead:

```js
placeOrder: async () => {
  await page.waitForNavigation( {
    waitUntil: 'networkidle0',
  } ),
  await expect( page ).toClick( '.wc-block-components-button__text', {
    text: 'Place Order',
  } ),
},
```

Or, this one:

```js
placeOrder: async () => {
await expect( page ).toClick( '.wc-block-components-button__text', {
    text: 'Place Order',
  } ),
  await page.waitForNavigation( {
    waitUntil: 'networkidle0',
  } ),
},
```

Both functions will have to wait for each promise to resolve sequentially, which will result in indefinitely waiting for a page to navigate to a new URL when the `page.waitForNavigation()` is executed.

You can test that using the following code:

This one fails with: `TimeoutError: Navigation timeout of 30000 ms exceeded` error ❌
```js
await page.waitForNavigation(); // The promise resolves after navigation has finished
await page.click( 'li.page-item-6 a' ); // Clicking the link will indirectly cause a navigation
```

And this one succeed ✅

```js
await Promise.all( [
	page.waitForNavigation(), // The promise resolves after navigation has finished
	page.click( 'li.page-item-6 a' ), // Clicking the link will indirectly cause a navigation
	// 'li.page-item-6 a' is the 'cart' page link on the navbar
] );
```